Makefile.am

```diff
@@ -158,14 +158,66 @@ base_files = \
 bin_PROGRAMS += csd
 
 csd_SOURCES = \
-	${base_files} \
+	${BASE_FILES} \
 	src/main.cpp
 
 csd_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS) $(GST_LIBS)
 
-.PHONY: samples
-samples:
-	@$(MAKE) -C samples
+#Samples
+EXTRA_PROGRAMS = samples/camera-sample-custom samples/camera-sample-client samples/camera-sample-mavlink-client
+
+if HAVE_REALSENSE
+EXTRA_PROGRAMS += samples/camera-sample-realsense
+endif
+
+samples_camera_sample_custom_LDADD = ${csd_LDADD}
+samples_camera_sample_realsense_LDADD = ${csd_LDADD}
+samples_camera_sample_client_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS)
+samples_camera_sample_mavlink_client_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS)
+
+samples_camera_sample_custom_SOURCES = \
+	$(BASE_FILES) \
+	src/main.cpp \
+	samples/stream_builder_custom.h \
+	samples/stream_builder_custom.cpp \
+	samples/stream_custom.cpp \
+	samples/stream_custom.h
+
+samples_camera_sample_realsense_SOURCES = \
+	$(BASE_FILES) \
+	src/main.cpp \
+	samples/stream_builder_realsense.h \
+	samples/stream_builder_realsense.cpp \
+	samples/stream_realsense.cpp \
+	samples/stream_realsense.h
+
+samples_camera_sample_client_SOURCES = \
+	samples/main_sample_client.cpp \
+	src/log.cpp \
+	src/log.h \
+	src/glib_mainloop.cpp \
+	src/glib_mainloop.h \
+	src/mainloop.cpp \
+	src/mainloop.h
+
+samples_camera_sample_mavlink_client_SOURCES = \
+	samples/main_sample_mavlink_client.cpp \
+	src/log.cpp \
+	src/log.h \
+	src/glib_mainloop.cpp \
+	src/glib_mainloop.h \
+	src/mainloop.cpp \
+	src/mainloop.h \
+	src/pollable.cpp \
+	src/pollable.h \
+	src/socket.cpp \
+	src/socket.h \
+	src/util.c \
+	src/util.h
+
+.PHONY: samples samples2
+samples: include/mavlink/ardupilotmega/mavlink.h
+	$(MAKE) $(EXTRA_PROGRAMS)
```

Comment:
/me confused here. Why do you need to call make from inside make? Wouldn't it suffice to just have like the dependency like below?

```
samples: $(EXTRA_PROGRAMS)
```

The mavlink header dependency is rather a dep from each program... so shouldn't be here, too